src/Symfony/Component/DependencyInjection/Tests/Compiler/AutowiringPassTest.php

```diff
@@ -0,0 +1,277 @@
+        $this->assertEquals('f', (string) $container->getDefinition('g')->getArgument(0));
+        $this->assertEquals('f', (string) $container->getDefinition('g')->getArgument(1));
+    }
+
+    public function testCompleteExistingDefinition()
+    {
+        $container = new ContainerBuilder();
+
+        $container->register('b', __NAMESPACE__.'\B');
+        $container->register('f', __NAMESPACE__.'\F');
+        $container->register('h', __NAMESPACE__.'\H')->addArgument(new Reference('b'));
+
+        $pass = new AutowiringPass();
+        $pass->process($container);
+
+        $this->assertCount(2, $container->getDefinition('h')->getArguments());
+        $this->assertEquals('b', (string) $container->getDefinition('h')->getArgument(0));
+        $this->assertEquals('f', (string) $container->getDefinition('h')->getArgument(1));
+    }
+
+    public function testCompleteExistingDefinitionWithNotDefinedArguments()
+    {
+        $container = new ContainerBuilder();
+
+        $container->register('b', __NAMESPACE__.'\B');
+        $container->register('f', __NAMESPACE__.'\F');
+        $container->register('h', __NAMESPACE__.'\H')->addArgument('')->addArgument('');
+
+        $pass = new AutowiringPass();
+        $pass->process($container);
+
+        $this->assertCount(2, $container->getDefinition('h')->getArguments());
+        $this->assertEquals('b', (string) $container->getDefinition('h')->getArgument(0));
+        $this->assertEquals('f', (string) $container->getDefinition('h')->getArgument(1));
+    }
+
+    /**
+     * @expectedException \Symfony\Component\DependencyInjection\Exception\RuntimeException
+     * @expectedExceptionMessage Unable to autowire type "Symfony\Component\DependencyInjection\Tests\Compiler\CollisionInterface".
+     */
+    public function testTypeCollision()
+    {
+        $container = new ContainerBuilder();
+
+        $container->register('c1', __NAMESPACE__.'\CollisionA');
+        $container->register('c2', __NAMESPACE__.'\CollisionB');
+        $container->register('a', __NAMESPACE__.'\CannotBeAutowired');
+
+        $pass = new AutowiringPass();
+        $pass->process($container);
+    }
+
+    public function testWithTypeSet()
+    {
+        $container = new ContainerBuilder();
+
+        $container->register('c1', __NAMESPACE__.'\CollisionA');
+        $container->register('c2', __NAMESPACE__.'\CollisionB')->addType(__NAMESPACE__.'\CollisionInterface');
+        $container->register('a', __NAMESPACE__.'\CannotBeAutowired');
+
+        $pass = new AutowiringPass();
+        $pass->process($container);
+
+        $this->assertCount(1, $container->getDefinition('a')->getArguments());
+        $this->assertEquals('c2', (string) $container->getDefinition('a')->getArgument(0));
+    }
+
+    public function testCreateDefinition()
+    {
+        $container = new ContainerBuilder();
+
+        $container->register('coop_tilleuls', __NAMESPACE__.'\LesTilleuls');
```

Comment:
I hope you understand this comment right: I think that using the real name of your company in the tests is "not right". I'd prefer to keep using synthetic names for everything (`acme`, `foo`, `bar`, etc.)